src/LowercaseSentenceTokenizer.java

```diff
@@ -1,36 +1,61 @@
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Scanner;
 
 /**
- * A tokenizer that converts text input to lowercase and splits it 
+ * A tokenizer that converts text input to lowercase and splits it
  * into a list of tokens, where each token is either a word or a period.
  */
 public class LowercaseSentenceTokenizer implements Tokenizer {
   /**
-   * Tokenizes the text from the given Scanner. The method should 
+   * Tokenizes the text from the given Scanner. The method should
    * convert the text to lowercase and split it into words and periods.
    * Words are separated by spaces, and periods are treated as separate tokens.
    * 
    * For example:
    * If the input text is: "Hello world. This is an example."
-   * The tokenized output should be: ["hello", "world", ".", "this", "is", "an", "example", "."]
+   * The tokenized output should be: ["hello", "world", ".", "this", "is", "an",
+   * "example", "."]
    * 
-   * Notice that the text is converted to lowercase, and each period is treated as a separate token.
+   * Notice that the text is converted to lowercase, and each period is treated as
+   * a separate token.
    * 
-   * However, a period should only be considered a separate token if it occurs at the end
+   * However, a period should only be considered a separate token if it occurs at
+   * the end
    * of a word. For example:
    * 
    * If the input text is: "Hello world. This is Dr.Smith's example."
-   * The tokenized output should be: ["hello", "world", ".", "this", "is", "dr.smith's", "example", "."]
+   * The tokenized output should be: ["hello", "world", ".", "this", "is",
+   * "dr.smith's", "example", "."]
    * 
-   * The internal period in Dr.Smith's is not treated as its own token because it does not occur at the end of the word.
+   * The internal period in Dr.Smith's is not treated as its own token because it
+   * does not occur at the end of the word.
    * 
    * @param scanner the Scanner to read the input text from
    * @return a list of tokens, where each token is a word or a period
    */
   public List<String> tokenize(Scanner scanner) {
-    // TODO: Implement this function to convert the scanner's input to a list of words and periods
-    return null;
+    // TODO: Implement this function to convert the scanner's input to a list of
+    // words and periods
+    List<String> convertedSentance = new ArrayList<>();
+    while (scanner.hasNext()) {
+      // Target each word in a given sentence
+      String word = scanner.next();
+
+      if (word.charAt(word.length() - 1) == '.') {
+        word.toLowerCase();
```

Comment:
This unfortunately doesn't quite work. When you call `toLowerCase` it actually doesn't modify the original string - it returns a new version of the String that's lowercase. With this code, `word` stays the same. You need `word = word.toLowerCase()` to reassign `word` to the new value. A test case that exposes this would involve a token like `sPoNgE.` (note the capitalization and the period)